js/modules/k6/grpc/client_test.go

```diff
@@ -761,3 +766,101 @@ func TestDebugStat(t *testing.T) {
 		})
 	}
 }
+
+func TestResolveFileDescriptors(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name                string
+		pkgs                []string
+		services            []string
+		expectedDescriptors int
+	}{
+		{
+			name:                "SuccessSamePackage",
+			pkgs:                []string{"mypkg"},
+			services:            []string{"Service1", "Service2", "Service3"},
+			expectedDescriptors: 3,
+		},
+		{
+			name:                "SuccessMultiPackages",
+			pkgs:                []string{"mypkg1", "mypkg2", "mypkg3"},
+			services:            []string{"Service1", "Service2", "Service3"},
+			expectedDescriptors: 3,
+		},
+		{
+			name:                "DeduplicateServices",
+			pkgs:                []string{"mypkg1"},
+			services:            []string{"Service1", "Service2", "Service1"},
+			expectedDescriptors: 2,
+		},
+		{
+			name:                "NoServices",
+			services:            []string{},
+			expectedDescriptors: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			var (
+				lsr  = &reflectpb.ListServiceResponse{}
+				mock = &getServiceFileDescriptorMock{}
+			)
+			for i, service := range tt.services {
+				// if only one package is defined then
+				// the package is the same for every service
+				pkg := tt.pkgs[0]
+				if len(tt.pkgs) > 1 {
+					pkg = tt.pkgs[i]
+				}
```

Comment:
I think this implies that the following would work just as well:
```suggestion
				// if only one package is defined then
				// the package is the same for every service
				pkg := tt.pkgs[0]
```
Just in case, I've already checked :slightly_smiling_face: The test successfully passes so you can leave only the 1st element in `pkgs`. From that follows that you can simplify `pkgs` field down to one `string`. This is true for all test cases here, in `TestResolveFileDescriptors`.

But at the moment I'm not fully clear that this is the intended logic: i.e. if there _can be_ a case with multiple services and multiple packages, it must be reflected in the test cases in such a way that it would fail without multiple packages.
Could you clarify the logic / expectations / test cases here?

Reply:
Thanks for catching it, I pushed a fixup for making the test fail without that condition. Let me know if it is clear or something is missing :+1: